src/configs/javascript.ts

```diff
@@ -116,6 +116,8 @@ export function javascript(options: OptionsIsInEditor & OptionsOverrides = {}):
           'DebuggerStatement',
           'LabeledStatement',
           'WithStatement',
+          'TSEnumDeclaration[const=true]',
```

Comment:
Banning TS syntax in JS config, thus we don't have to maintain two separated restricted syntax lists